rocket-lazy-load.php

```diff
@@ -536,11 +582,16 @@ function rocket_lazyload_iframes( $html ) {
  * @return string     Youtube video id or false if none found.
  */
 function rocket_lazyload_get_youtube_id_from_url( $url ) {
-	$pattern = '#^(?:https?://)?(?:www\.)?(?:youtu\.be/|youtube\.com(?:/embed/|/v/|/watch\?v=))([\w-]{11})#iU';
+	$pattern = '#^(?:https?://)?(?:www\.)?(?:(?:youtu\.be/|youtube\.com|youtube-nocookie\.com)(?:/embed/|/v/|/watch\?v=))([\w-]{11})#iU';
```

Comment:
Fix: removed a trailing slash after `youtu\.be`.
Simpler: removed the `(?:XXX)` around the whole `youtube.com/embed` part (useless).
Improvement: the `embed` group.
[Tests](https://regex101.com/r/KF8xvI/1).
```php
$pattern = '#^(?:https?://)?(?:www\.)?(?:youtu\.be|youtube\.com|youtube-nocookie\.com)/(?:embed/|v/|watch/?\?v=)([\w-]{11})#iU';
```
I'm also concerned about the protocol. I think it should be `(?:https?:)?//`.